### ISSUE TITLE
[question] Startup hangs in Jenkins

### ISSUE THREAD

**Issue by ismailkuruca** [testcontainers/testcontainers-node#61]
I have been using testcontainers-java for the past couple of year and I love it. Hence I tried to give it a shot to node version (2.7.0) of it.

I have a fairly simple test setup:
```typescript
jest.setTimeout(600000)
let pgContainer: any

beforeAll(async () => {
  pgContainer = await new GenericContainer('postgres')
    .withEnv('POSTGRES_USER', 'test')
    .withEnv('POSTGRES_PASSWORD', 'test')
    .withEnv('POSTGRES_DB', 'postgres')
    //added for testing purposes
    //.withStartupTimeout(new Duration(200, TemporalUnit.SECONDS))
    .withExposedPorts(5432)
    .start()

  expect(pgContainer).toBeTruthy()
  console.log(pgContainer.boundPorts)
})

afterAll(async () => {
  await pgContainer.stop()
})

describe('verify it works', function() { ... }
```

which works perfectly on my local machine:

```
2020-04-06T13:12:54.372Z testcontainers INFO: Using Docker defaults
2020-04-06T13:12:54.452Z testcontainers INFO: Creating container for image: postgres:latest
2020-04-06T13:12:54.586Z testcontainers INFO: Starting container with ID: 6647ae7575ff97de3be4444b704bec49edb68ffc74275f6b5be7b8e7e1ff7f79
2020-04-06T13:12:55.238Z testcontainers DEBUG: Waiting for container to be ready
2020-04-06T13:12:55.238Z testcontainers DEBUG: Waiting for host port 63608
2020-04-06T13:12:55.240Z testcontainers DEBUG: Waiting for internal port 5432
2020-04-06T13:12:57.996Z testcontainers DEBUG: Container is ready
```

But when I try to run it on my company's Jenkins slave (which sadly I have very limited access to)
it fails to start
```
$ DEBUG=testcontainers yarn test:unit
$ jest --config test/config.unit.json
2020-04-06T13:02:22.935Z testcontainers INFO: Using Docker in Docker method
2020-04-06T13:02:22.961Z testcontainers INFO: Pulling image: postgres:latest
2020-04-06T13:02:29.102Z testcontainers INFO: Creating container for image: postgres:latest
2020-04-06T13:02:30.761Z testcontainers INFO: Starting container with ID: 8735ecbb25cd26c084db2e009e6553fed518b2446e3d9a9a5edef1f9aec1b14c
2020-04-06T13:02:31.566Z testcontainers DEBUG: Waiting for container to be ready
2020-04-06T13:02:31.567Z testcontainers DEBUG: Waiting for host port 44063
2020-04-06T13:02:31.568Z testcontainers DEBUG: Waiting for internal port 5432
FAIL src/middleware/test/index.spec.ts
  ● Test suite failed to run

    TypeError: Cannot read property 'stop' of undefined

      26 | 
      27 | afterAll(async () => {
    > 28 |   await pgContainer.stop()
         |                     ^
      29 | })
      30 | 
      31 | describe('verify it works', function() {

      at ../src/middleware/clientAuth/index.spec.ts:28:21
      at ../src/middleware/clientAuth/index.spec.ts:8:71
      at Object.<anonymous>.__awaiter (../src/middleware/clientAuth/index.spec.ts:4:12)
      at Object.<anonymous>.afterAll (../src/middleware/clientAuth/index.spec.ts:27:21)
```

Jenkins pipeline that is responsible to orchestrate is something like this (I use NodeJS plugin with Node version of 10.19)

```groovy
stage("test") {
                nodejs(nodeJSInstallationName: 'node_10') {
                    sh 'yarn && CI=1 yarn test'
                }   
            }
```

It looks like the similar issues that have been fixed in this repository but only thing that I can't make sense of is, this is pretty much the same setup I have in my Java project (which uses a Java plugin on Jenkins instead of a NodeJS one).

Is there any tip you could give me about further debugging this?
Thanks

**Comment by cristianrgreco** [comment]
Hi @ismailkuruca, is the Jenkins slave also running in a Docker container?

**Comment by ismailkuruca** [comment]
Hey @cristianrgreco yes, it is running in a Docker container

**Comment by ismailkuruca** [comment]
Hi again, I have managed to solve my issue by adding specialised healthchecks. Although I am not sure why it becomes a problem in CI environment and why local builds do not require such checks, would be great if you can give me some insights about that part.

For future reference, what I did was adding:
```typescript
.withHealthCheck({
        test: 'pg_isready -U postgres',
        interval: new Duration(2, TemporalUnit.SECONDS),
        timeout: new Duration(5, TemporalUnit.SECONDS),
        retries: 3,
      })
```

**Comment by cristianrgreco** [comment]
@ismailkuruca Since this issue was raised we added container startup logging, so we'd be able to see what's happening inside the container. I don't suppose you could re-run your test on the Jenkins slave with DEBUG=testcontainers and share the output?

**Comment by ismailkuruca** [comment]
Hey @cristianrgreco , just re-tried with 2.17,
It still fails when I dont give it a proper Wait strategy (unlike a local machine where this is not needed)

Here is the code I am using:
```javascript
await new GenericContainer('postgres')
      .withEnv('POSTGRES_USER', 'test')
      .withEnv('POSTGRES_PASSWORD', 'test')
      .withEnv('POSTGRES_DB', 'postgres')
      .withExposedPorts(5432)
      .start()
```

```+ yarn test:integration
yarn run v1.22.4
$ DEBUG=testcontainers jest --runInBand --testPathPattern=integration.test.js
2020-07-07T10:06:53.319Z testcontainers INFO: Using Docker in Docker method
2020-07-07T10:06:53.337Z testcontainers INFO: Pulling image: postgres:latest
2020-07-07T10:07:00.557Z testcontainers INFO: Creating container for image: postgres:latest
2020-07-07T10:07:02.719Z testcontainers INFO: Starting container with ID: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3
2020-07-07T10:07:03.436Z testcontainers DEBUG: Waiting for container to be ready
2020-07-07T10:07:03.437Z testcontainers DEBUG: Waiting for host port 44775
2020-07-07T10:07:03.437Z testcontainers DEBUG: Waiting for internal port 5432
2020-07-07T10:07:03.466Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: NThe files belonging to this database system will be owned by user "postgres".
2020-07-07T10:07:03.466Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: ,This user must also own the server process.
2020-07-07T10:07:03.466Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: CThe database cluster will be initialized with locale "en_US.utf8".
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: BThe default database encoding has accordingly been set to "UTF8".
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: @The default text search configuration will be set to "english".
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: "Data page checksums are disabled.
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: Ifixing permissions on existing directory /var/lib/postgresql/data ... ok
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: creating subdirectories ... ok
2020-07-07T10:07:03.467Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 9selecting dynamic shared memory implementation ... posix
2020-07-07T10:07:03.478Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: *selecting default max_connections ... 100
2020-07-07T10:07:03.498Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: +selecting default shared_buffers ... 128MB
2020-07-07T10:07:03.513Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: (selecting default time zone ... Etc/UTC
2020-07-07T10:07:03.514Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: $creating configuration files ... ok
2020-07-07T10:07:03.621Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3:  running bootstrap script ... ok
2020-07-07T10:07:04.066Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 0performing post-bootstrap initialization ... ok
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: Ginitdb: warning: enabling "trust" authentication for local connections
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: FYou can change this by editing pg_hba.conf or using the option -A, or
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: <--auth-local and --auth-host, the next time you run initdb.
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: syncing data to disk ... ok
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 6Success. You can now start the database server using:
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 8    pg_ctl -D /var/lib/postgresql/data -l logfile start
2020-07-07T10:07:04.117Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:04.141Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: �waiting for server to start....2020-07-07 10:07:04.140 UTC [66] LOG:  starting PostgreSQL 12.3 (Debian 12.3-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
2020-07-07T10:07:04.141Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: d2020-07-07 10:07:04.141 UTC [66] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2020-07-07T10:07:04.153Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: `2020-07-07 10:07:04.152 UTC [67] LOG:  database system was shut down at 2020-07-07 10:07:04 UTC
2020-07-07T10:07:04.155Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: V2020-07-07 10:07:04.155 UTC [66] LOG:  database system is ready to accept connections
2020-07-07T10:07:04.217Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3:  done
2020-07-07T10:07:04.217Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: server started
2020-07-07T10:07:04.217Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:04.217Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: L/usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
2020-07-07T10:07:04.217Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:04.218Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: F2020-07-07 10:07:04.218 UTC [66] LOG:  received fast shutdown request
2020-07-07T10:07:04.219Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: kwaiting for server to shut down....2020-07-07 10:07:04.218 UTC [66] LOG:  aborting any active transactions
2020-07-07T10:07:04.219Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: y2020-07-07 10:07:04.219 UTC [66] LOG:  background worker "logical replication launcher" (PID 73) exited with exit code 1
2020-07-07T10:07:04.219Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 52020-07-07 10:07:04.219 UTC [68] LOG:  shutting down
2020-07-07T10:07:04.224Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: D2020-07-07 10:07:04.224 UTC [66] LOG:  database system is shut down
2020-07-07T10:07:04.318Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3:  done
2020-07-07T10:07:04.318Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: server stopped
2020-07-07T10:07:04.318Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:04.318Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 6PostgreSQL init process complete; ready for start up.
2020-07-07T10:07:04.318Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: 
2020-07-07T10:07:04.327Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: �2020-07-07 10:07:04.327 UTC [1] LOG:  starting PostgreSQL 12.3 (Debian 12.3-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
2020-07-07T10:07:04.328Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: U2020-07-07 10:07:04.327 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2020-07-07T10:07:04.328Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: P2020-07-07 10:07:04.327 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2020-07-07T10:07:04.328Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: c2020-07-07 10:07:04.328 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2020-07-07T10:07:04.337Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: `2020-07-07 10:07:04.337 UTC [80] LOG:  database system was shut down at 2020-07-07 10:07:04 UTC
2020-07-07T10:07:04.340Z testcontainers TRACE: b9b8e2ca81577fa07e70daa9f5e8977afa51865987de0ee50ec12d476a13c5f3: U2020-07-07 10:07:04.340 UTC [1] LOG:  database system is ready to accept connections
  console.log
    Error: Port 44775 not bound after 60000ms
        at /home/jenkins/agent/workspace/less_multibranch_pr_CBB-111-test/node_modules/testcontainers/dist/wait-strategy.js:59:23
        at IntervalRetryStrategy.<anonymous> (/home/jenkins/agent/workspace/less_multibranch_pr_CBB-111-test/node_modules/testcontainers/dist/retry-strategy.js:35:28)
        at Generator.next (<anonymous>)
        at fulfilled (/home/jenkins/agent/workspace/less_multibranch_pr_CBB-111-test/node_modules/testcontainers/dist/retry-strategy.js:4:58)
        at processTicksAndRejections (internal/process/task_queues.js:82:5)

      at createPGContainer (node_modules/support/testUtil/testContainersUtil.js:15:13)
```

If I add one of the proper Wait strategies like
```javascript
      .withWaitStrategy(Wait.forLogMessage('database system is ready to accept connections'))
//or
      .withWaitStrategy(Wait.forHealthCheck())
      .withHealthCheck({
        test: 'pg_isready -U postgres',
        interval: new Duration(2, TemporalUnit.SECONDS),
        timeout: new Duration(5, TemporalUnit.SECONDS),
        retries: 3,
      })
```

it works perfectly.

**Comment by cristianrgreco** [comment]
Thanks @ismailkuruca, could you try the latest 2.18.3? I've upgraded dockerode/docker-modem to their latest versions

**Comment by ismailkuruca** [comment]
Hi again @cristianrgreco 
Here is the log with 2.18.3

AFAIK the slaves are running inside AmazonLinux2 AMIs.

```
$ DEBUG=testcontainers jest --runInBand --testPathPattern=integration.test.js
2020-07-07T13:31:10.897Z testcontainers INFO: Using Docker in Docker method
2020-07-07T13:31:10.948Z testcontainers INFO: Pulling image: postgres:latest
2020-07-07T13:31:18.467Z testcontainers INFO: Creating container for image: postgres:latest
2020-07-07T13:31:18.950Z testcontainers INFO: Starting container with ID: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9
2020-07-07T13:31:19.866Z testcontainers DEBUG: Waiting for container to be ready
2020-07-07T13:31:19.866Z testcontainers DEBUG: Waiting for host port 32963
2020-07-07T13:31:19.868Z testcontainers DEBUG: Waiting for internal port 5432
2020-07-07T13:31:19.948Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: NThe files belonging to this database system will be owned by user "postgres".
2020-07-07T13:31:19.948Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: ,This user must also own the server process.
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: CThe database cluster will be initialized with locale "en_US.utf8".
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: BThe default database encoding has accordingly been set to "UTF8".
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: @The default text search configuration will be set to "english".
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: "Data page checksums are disabled.
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: Ifixing permissions on existing directory /var/lib/postgresql/data ... ok
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: creating subdirectories ... ok
2020-07-07T13:31:19.949Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 9selecting dynamic shared memory implementation ... posix
2020-07-07T13:31:19.970Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: *selecting default max_connections ... 100
2020-07-07T13:31:19.990Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: +selecting default shared_buffers ... 128MB
2020-07-07T13:31:20.017Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: (selecting default time zone ... Etc/UTC
2020-07-07T13:31:20.017Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: $creating configuration files ... ok
2020-07-07T13:31:20.163Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9:  running bootstrap script ... ok
2020-07-07T13:31:20.910Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 0performing post-bootstrap initialization ... ok
2020-07-07T13:31:21.004Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: syncing data to disk ... ok
2020-07-07T13:31:21.004Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:21.004Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:21.004Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 6Success. You can now start the database server using:
2020-07-07T13:31:21.004Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:21.004Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 8    pg_ctl -D /var/lib/postgresql/data -l logfile start
2020-07-07T13:31:21.004Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:21.004Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: Ginitdb: warning: enabling "trust" authentication for local connections
2020-07-07T13:31:21.005Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: FYou can change this by editing pg_hba.conf or using the option -A, or
2020-07-07T13:31:21.005Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: <--auth-local and --auth-host, the next time you run initdb.
2020-07-07T13:31:21.138Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: �waiting for server to start....2020-07-07 13:31:21.132 UTC [67] LOG:  starting PostgreSQL 12.3 (Debian 12.3-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
2020-07-07T13:31:21.138Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: d2020-07-07 13:31:21.132 UTC [67] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2020-07-07T13:31:21.151Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: `2020-07-07 13:31:21.150 UTC [68] LOG:  database system was shut down at 2020-07-07 13:31:20 UTC
2020-07-07T13:31:21.160Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: V2020-07-07 13:31:21.154 UTC [67] LOG:  database system is ready to accept connections
2020-07-07T13:31:21.200Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9:  done
2020-07-07T13:31:21.200Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: server started
2020-07-07T13:31:21.200Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:21.200Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: L/usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
2020-07-07T13:31:21.200Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:21.217Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: F2020-07-07 13:31:21.213 UTC [67] LOG:  received fast shutdown request
2020-07-07T13:31:21.222Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: kwaiting for server to shut down....2020-07-07 13:31:21.218 UTC [67] LOG:  aborting any active transactions
2020-07-07T13:31:21.223Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: y2020-07-07 13:31:21.221 UTC [67] LOG:  background worker "logical replication launcher" (PID 74) exited with exit code 1
2020-07-07T13:31:21.223Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 52020-07-07 13:31:21.221 UTC [69] LOG:  shutting down
2020-07-07T13:31:21.232Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: D2020-07-07 13:31:21.227 UTC [67] LOG:  database system is shut down
2020-07-07T13:31:21.320Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9:  done
2020-07-07T13:31:21.320Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: server stopped
2020-07-07T13:31:21.320Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:21.320Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 6PostgreSQL init process complete; ready for start up.
2020-07-07T13:31:21.320Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: 
2020-07-07T13:31:21.353Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: �2020-07-07 13:31:21.351 UTC [1] LOG:  starting PostgreSQL 12.3 (Debian 12.3-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
2020-07-07T13:31:21.353Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: U2020-07-07 13:31:21.352 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2020-07-07T13:31:21.353Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: P2020-07-07 13:31:21.352 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2020-07-07T13:31:21.353Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: c2020-07-07 13:31:21.352 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2020-07-07T13:31:21.376Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: `2020-07-07 13:31:21.368 UTC [76] LOG:  database system was shut down at 2020-07-07 13:31:21 UTC
2020-07-07T13:31:21.386Z testcontainers TRACE: f0657ede10b8f4c220935aa1cc2b3140abaa6d2639240b4c7cff22b37b2a0cf9: U2020-07-07 13:31:21.385 UTC [1] LOG:  database system is ready to accept connections
  console.log
    Error: Port 32963 not bound after 60000ms
```

**Comment by cristianrgreco** [comment]
What's the version of Docker running on the slave and what's the contents of `/.dockerenv`?

**Comment by cristianrgreco** [comment]
Could you please share the output of `docker network inspect bridge` and confirm that the file `/.dockerenv` exists?

**Comment by cristianrgreco** [comment]
I've merged a change to master which re-works how the host is resolved when running inside a container. Are you able to add a dependency to this GitHub repository in your `package.json` to test it? You should be able to do:
```json
"testcontainers": "testcontainers/testcontainers-node",
```

**Comment by cristianrgreco** [comment]
@ismailkuruca Were you able to give it a try? I'm optimistic the version on master will fix your problem

**Comment by cristianrgreco** [comment]
This has now been released in version 2.19.0. Let me know if it fixes it for you.

**Comment by oleksandrz-tt** [comment]
You may need to run your jenkins with --net=host option if it's running in docker
